native/core/src/execution/shuffle/shuffle_writer.rs

```diff
@@ -572,15 +567,12 @@ impl ShuffleRepartitioner {
             output_data.write_all(&output_batches[i])?;
             output_batches[i].clear();
 
-            // append partition in each spills
-            for spill in &output_spills {
-                let length = spill.offsets[i + 1] - spill.offsets[i];
-                if length > 0 {
-                    let mut spill_file =
-                        BufReader::new(File::open(spill.file.path()).map_err(Self::to_df_err)?);
-                    spill_file.seek(SeekFrom::Start(spill.offsets[i]))?;
-                    std::io::copy(&mut spill_file.take(length), &mut output_data)
-                        .map_err(Self::to_df_err)?;
+            if let Some(spill_data) = self.buffered_partitions[i].spill_file.as_ref() {
```

Comment:
This is basically saying, if 1) We have a SpillFile, and 2) the length of that SpillFile is greater than 0 -> we need to copy that spilled data to the output buffer. My question is: because we're now reusing spill files instead of creating them for each spill event, when does the reused SpillFile get truncated back to 0 now that we've copied all of the data to `output_data`? It's very likely it's happening somewhere that I don't see, so perhaps a comment here where that happens.

Reply:
Yes, that is correct. We never truncate the spill file. We just append to it. At the end of the shuffle, we copy the contents of each spill file to the shuffle file. I will add some comments to make this clearer.

Reply:
I added some comments and also removed the check for length > 0 since that was redundant (we only create the spill file if we have data to spill)

Reply:
Makes sense. I misunderstood the granularity at which the spill file could be written to the output file.